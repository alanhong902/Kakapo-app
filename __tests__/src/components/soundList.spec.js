import React from 'react';
import { SoundList } from 'components/';
import { getData, createComponentWithIntl, randomSounds } from '../helper';

test('<SoundList/> render empty', () => {
  const props = {
    ...getData('sounds'),
    ...getData('themes'),
  };
  const tree = createComponentWithIntl(<SoundList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<SoundList/> render', () => {
  const props = {
    ...getData('sounds'),
    ...getData('themes'),
    sounds: randomSounds(4),
  };
  const tree = createComponentWithIntl(<SoundList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<SoundList/> progress === 1', () => {
  const props = {
    ...getData('sounds'),
    ...getData('themes'),
    sounds: randomSounds(8),
  };
  const tree = createComponentWithIntl(<SoundList {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
