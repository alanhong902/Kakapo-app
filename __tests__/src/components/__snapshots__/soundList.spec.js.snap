// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundList/> progress === 1 1`] = `
<section>
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;

exports[`<SoundList/> render 1`] = `
<section>
  <div
    className="sound-list"
  />
  <div
    className="sound-list"
  >
    <div
      className="sound-item-wrap"
    >
      <div
        className="item editing"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue="thunder"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="name"
            >
              <span>
                Name
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <div
            className="group"
          >
            <input
              autoComplete="off"
              defaultValue={null}
              name="tags"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <label
              htmlFor="tags"
            >
              <span>
                Tag
              </span>
            </label>
            <span
              className="highlight"
            />
            <span
              className="bar"
            />
          </div>
          <a
            className="button"
            onClick={[Function]}
            role="link"
            tabIndex={-1}
          >
            <span>
              Cancel
            </span>
          </a>
          <button
            className="button"
            style={
              Object {
                "backgroundColor": "#4CAF50",
                "borderColor": "#4CAF50",
              }
            }
            type="button"
          >
            <span>
              Save
            </span>
          </button>
        </form>
      </div>
    </div>
  </div>
</section>
`;

exports[`<SoundList/> render empty 1`] = `<div />`;
